test/typescript/sample-usage.tsx

```diff
@@ -14,26 +15,44 @@ const testArr: ITodo[] = [
   { text: 'mountain', id: '11' },
   { text: 'air', id: '8' },
   { text: 'plants', id: '9' },
+  { text: 'air', id: '8' },
+  { text: 'air', id: '8' },
 ];
 
-type IProps = {
+type IChildProps = {
   todo: ITodo;
 };
 
-class TestChild extends React.Component<IProps> {
-  shouldComponentUpdate(nextProps: IProps) {
+class TestChild extends React.Component<IChildProps> {
+  shouldComponentUpdate(nextProps: IChildProps) {
     return !equal(this.props, nextProps);
   }
 
   render() {
     const todo = this.props.todo;
+
     return <div>{todo.text}</div>;
   }
 }
 
-class TestContainer extends React.Component {
+type IContainerState = { completed: ITodo[] };
+
+class TestContainer extends React.Component<{}, IContainerState> {
```

Comment:
Let's have a completely new component (or set of components, if necessary) for this test. I don't want our components to do "double duty" by trying to test multiple things. These components sure be single-minded, focused on testing our types against react-redux